lib/webdriveragent.js

```diff
@@ -286,6 +308,25 @@ class WebDriverAgent {
       return await this.getStatus();
     }
 
+    if (this.isRealDevice && this.usePreinstalledWDA) {
```

Comment:
I think we should verify it in xcuitest driver if usePreinstalledWDA cap is set to true and a simulator is selected. In such case a validation error must be thrown

Reply:
I find the current method a bit to long. Perhaps, it makes sense to move this code block into a separate method?

Reply:
ah, it makes sense